gfsphysics/GFS_layer/GFS_physics_driver.F90

```diff
@@ -5484,10 +5480,10 @@ subroutine GFS_physics_driver                                 &
 
       if (Model%cplflx .or. Model%cplchm) then
         do i = 1, im
-          Coupling%rain_cpl(i) = Coupling%rain_cpl(i) &
-                               + Diag%rain(i) * (one-Sfcprop%srflag(i))
-          Coupling%snow_cpl(i) = Coupling%snow_cpl(i) &
-                               + Diag%rain(i) * Sfcprop%srflag(i)
+          Tbd%drain_cpl(i)= Diag%rain(i) * (one-Sfcprop%srflag(i))
```

Comment:
Dom,

I just tried to build this branch in S2S-model on Cheyenne and got a seg fault at this line. Should there be a "do_sppt" statement here? log file :/glade/scratch/worthen/hour2/log.out

Reply:
Hmm. @pjpegion please help. I see in `GFS_typedefs.F90`:
```
...
    if (Model%do_sppt) then
      allocate (Tbd%dtdtr     (IM,Model%levs))
      allocate (Tbd%dtotprcp  (IM))
      allocate (Tbd%dcnvprcp  (IM))
      allocate (Tbd%drain_cpl (IM))
      allocate (Tbd%dsnow_cpl (IM))
...
```
Since the updates of Coupling%rain_cpl and Coupling%snow_cpl need to be applied if Model%cplflx and/or Model%cplchm is true as well, I would assume that we have to allocate the fields in GFS_typedefs in this case as well.

Reply:
I just pushed an update to ccpp-physics (doc changes only) and to fv3atm (allocation of arrays).

Reply:
We don't run the stochastic physics package in the coupled model, so do_sppt is false.

Reply:
In any case, with this code change you need the arrays allocated if cplchm or cplflx (or both) are true, independent of whether do_sppt is true or false.

Reply:
OK, thanks.

Reply:
We do have some who run stochastic physics in the coupled model.

Reply:
Yes, and this will work regardless of whether do_sppt is .true. or .false.

Reply:
Need to get Phil and GEFS group in to create a stochy test in coupled model, we have one on ufs-weather-model.

Reply:
@DeniseWorthen you are correct.  I only ran regression tests in the atmosphere model, and in the coupled model only with stochastic_physics.  I will fix this issue.

Reply:
... it is already fixed ... see commit https://github.com/NOAA-EMC/fv3atm/pull/86/commits/6f4cb56c6ab1c3401e5033c7c854aa382574288e